### DISCUSSION TITLE
How to handle token expired when user switch from background ?

### DISCUSSION THREAD

**Discussion by florian-mlr** [supabase-community/supabase-kt#553] [Q&A]
My token has a validity period of one hour but it may be that the user put the application in background and when he returns, the token has expired, so all calls launched at the moment the screen appears fail .
I would perhaps like to do a token check before each call requiring a token, is there an adapter or interceptor that allows this ?

**Comment by jan-tennert** [comment]
If you are on Android, then the `Auth#sessionStatus` flow should be `LoadingFromStorage` until the session gets refreshed and the `sessionStatus` updated to `Authenticated`

**Reply by florian-mlr** [reply]
I tried to follow this logic but the thing is that each time the user returns from the background, the state changes from loading from storage to authenticated which means that my compose views lose their state, for example if I had navigated from screen A to B, I return to screen A.
I would have preferred an interceptor system which checks the token before each call.

**Reply by jan-tennert** [reply]
How do you imagine this token check? Do requests silently fail if the session isn't refreshed yet? Or they just suspend the requests until refreshing?

**Reply by florian-mlr** [reply]
Yes i think that i would suspend the requests until refreshing

**Reply by jan-tennert** [reply]
I would just check for the session status and show a loading screen in case the session isn't refreshed, but if you really don't want this, you could check the `UserSession#expiresAt` property to see if the session is expired.

**Reply by florian-mlr** [reply]
I think the first solution is not very adapted to my case, i have background tasks in my app, sometimes the user not launch the app but the background task should be executed, if the token expired then the background would fail.

**Reply by jan-tennert** [reply]
Yea, then you have to check for the expiration yourself. You can also take a look at https://github.com/supabase-community/supabase-kt/issues/489

**Reply by florian-mlr** [reply]
Thank you, in my case I think that if I would have created a client I would use https://ktor.io/docs/client-bearer-auth.html#configure to manage the token, what do you think ?

**Reply by jan-tennert** [reply]
No, you don't have to manage auth on your own, just disable the config property as described in the issue and start auto refresh yourself. I'm still not sure how you would suspend until the session is refreshed without using sessionStatus
